src/Sidebar/Sidebar.tsx

```diff
@@ -8,12 +7,43 @@ import {
   SidebarHeaderStyles,
 } from "./sidebar.style";
 
-const renderableChildrenPropType = PropTypes.oneOfType([
-  PropTypes.arrayOf(PropTypes.node),
-  PropTypes.node,
-]);
+type LinkProps = {
+  iconName: string;
+  active?: boolean;
+  href: string;
+  children?: ReactNode;
+};
+
+type HeaderProps = {
+  logoUrl?: string;
+  logoAltText?: string;
+  href?: string;
+  children?: ReactNode;
+};
+
+type SideBarProps = {
+  children: ReactNode;
+};
+
+interface LinkFC<T> extends React.FC<T> {
+  displayName: string;
+}
 
-const Link = ({ iconName, active, href, children }) => {
+interface HeaderFC<T> extends React.FC<T> {
```

Comment:
no need fot this extraction let's use `FC` directly in the component

Reply:
optional comment

Reply:
I just made a generic type to give types to `displayName`, `Header`, `Link` and its nothing fancy actually, but how do you think to avoid these ?